v2/tools/generator/internal/config/glob_matcher.go

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package config
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+	"github.com/pkg/errors"
+)
+
+// A globMatcher is used to match a string against a literal or wildcard
+type globMatcher struct {
+	glob       string         // The glob we're matching, may contain wildcards * and ?
+	regex      *regexp.Regexp // A regular expression to match the glob
+	once       sync.Once
+	matched    bool
+	candidates set.Set[string]
+}
+
+var _ StringMatcher = &globMatcher{}
+
+// newGlobMatcher returns a new matcher for handling wildcards
+func newGlobMatcher(glob string) *globMatcher {
+	// Guard against misuse
+	if !HasWildCards(glob) {
+		msg := fmt.Sprintf("glob string %q has no wildcards", glob)
+		panic(msg)
+	}
+
+	return &globMatcher{
+		glob:       glob,
+		candidates: make(set.Set[string]),
+	}
+}
+
+func (gm *globMatcher) Matches(term string) bool {
+	gm.once.Do(gm.createRegex)
```

Comment:
Why defer creating the regex until use, rather than creating it in `newGlobMatcher`? It doesn't seem like there's going to be a situation where the regex isn't needed, and creating it later makes it _slightly_ harder to track down failures if there's a panic from `MustCompile`.

Reply:
Ooh, I just realised - is this to do with making a zero `globMatcher` work like in a new `TypeMatcher`? That's fair enough then.

Reply:
Yup. If the value is never set, the zero value needs to work.